prow/cmd/deck/static/tide-history/tide-history.ts

```diff
@@ -0,0 +1,310 @@
+import {cell} from "../common/common";
+import {JobState} from "../api/prow";
+import {HistoryData, Record, PRMeta} from "../api/tide-history";
+
+declare const tideHistory: HistoryData;
+
+// http://stackoverflow.com/a/5158301/3694
+function getParameterByName(name: string): string | null {
+  const match = RegExp('[?&]' + name + '=([^&/]*)').exec(
+    window.location.search);
+  return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
+}
+
+interface Options {
+  repos: {[key: string]: boolean};
+  branchs: {[key: string]: boolean};
+  actions: {[key: string]: boolean};
+  states: {[key: string]: boolean};
+  authors: {[key: string]: boolean};
+  pulls: {[key: string]: boolean};
+}
+
+function optionsForRepoBranch(repo: string, branch: string): Options {
+  const opts: Options = {
+    repos: {},
+    branchs: {},
+    actions: {},
+    states: {},
+    authors: {},
+    pulls: {},
+  };
+
+  const hist: {[key: string]: Record[]} = typeof tideHistory !== 'undefined' ? tideHistory.History : {};
```

Comment:
I don't think we handle this anywhere else. There isn't a super elegant way to deal with it, given our somewhat awkward approach to loading in data via `<script>` tags. We can either punt and add to the todo list for cleaning up data loading everywhere else, or in the interim you could do something like this in the template:

```html
<script type="text/javascript">
var tideHistory = undefined;
</script>
<script type="text/javascript" src="/tide-history.js?var=tideHistory">
```

Then you can reasonably change the declaration of `tideHistory` to be of type `HistoryData | undefined`, and then test for its existence before doing any real work. Then you can report the missing data — perhaps add a hidden-by-default error report element above the table and show it if something's broken. We won't have any sort of useful error message until we clean up the data loading, so it'd have to be something relatively generic.

Reply:
> We can either punt and add to the todo list for cleaning up data loading everywhere else, or in the interim you could do something like this in the template

Since this could affect every Prow page, I'll punt for now and we can fix this generically later. I don't this has ever actually been an issue in practice.